src/ficamp/__main__.py

```diff
@@ -125,6 +136,9 @@ def categorize(engine):
             print(f"Got {len(results)} Tx to categorize")
             for tx in results:
                 print(f"Processing {tx}")
+                hinted_category = infer_tx_category(tx)
```

Comment:
this should only happen if user opts in to `--hints`.
Otherwise this breaks usage without a google API key.